inject/src/main/java/io/micronaut/inject/BeanDefinition.java

```diff
@@ -166,6 +157,13 @@ default boolean isIterable() {
         return hasDeclaredStereotype(EachProperty.class) || hasDeclaredStereotype(EachBean.class);
     }
 
+    /**
+     * @return Is the type configuration properties.
+     */
+    default boolean isConfigurationProperties() {
+        return isIterable() || hasDeclaredStereotype(ConfigurationReader.class);
```

Comment:
This might not be correct: `EachBean` has nothing to do with configuration properties.

Reply:
True but the existing logic in `BeanDefinitionWriter` which computes this value for `BeanDefinitionReference` has the same logic. See https://github.com/micronaut-projects/micronaut-core/blob/4.0.x/core-processor/src/main/java/io/micronaut/inject/writer/BeanDefinitionWriter.java#L4093-L4095

Reply:
I feel like we would rely on this check by mistake

Reply:
I agree but it has been that way for some time 🤔